packages/nitro-protocol/src/index.ts

```diff
@@ -1,4 +1,11 @@
 import {Signature} from 'ethers/utils';
+import NitroAdjudicatorArtifact from '../build/contracts/NitroAdjudicator.json';
```

Comment:
This approach, with the current .gitignore, means this package will be broken unless `yarn build` is run, since these files don't exist.

I'm not sure if this is the best approach:
- it's appealing _not_ to check in an artifact that's a pure function of the contracts (which are checked in)
- it's _not_ appealing that the project is broken (as in `npx tsc` fails) when I first clone the repo

If we remove `../build/contracts` from .gitignore, I recommend scrubbing the artifacts from anything other than: `abi`, `contractName` and `bytecode`

Reply:
@tomclose and I had a quick chat about this and he raised a valid point about treating the contract files similar to our typescript files, as in they both must be compiled before the package can be used by another package.

> * it's _not_ appealing that the project is broken (as in `npx tsc` fails) when I first clone the repo

I think our approach here is that `npx tsc` is not considered "building" `nitro-protocol`. We expect people to run `yarn build` if they want to build the `nitro-protocol` package.

The advantage of this is that it allows consumers of `nitro-protocol` to not have to worry about compiling solidity files. 

Of course the downside is that building `nitro-protocol` now has the overhead of compiling the contracts.

Reply:
Yeah, I'm in favour of **not** checking them in. If you change your notion of what "broken" means, then it also seems fine to not check them in.

I think we're better off punting this issue, but before publishing to npm, we probably want to run some "production` build step, that
1. compiles the contracts
2. strip out unnecessary data from the artifacts
3. record the deployed addresses in a (published) network context file